packages/edit-post/src/utils/find-template.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Browser dependencies
+ */
+const { fetch } = window;
+
+/**
+ * Find the template for a given page path.
+ *
+ * @param {string}   path The page path.
+ * @param {Function} getEntityRecords The promise-returning `getEntityRecords` selector to use.
+ *
+ * @return {number} The found template ID.
+ */
+export default async function findTemplate( path, getEntityRecords ) {
+	const { data } = await fetch(
+		addQueryArgs( findTemplate.siteUrl + path, {
+			'_wp-find-template': true,
+		} )
+	).then( ( res ) => res.json() );
+
+	let newTemplate = data;
+	if ( newTemplate === null ) {
+		newTemplate = (
+			await getEntityRecords( 'postType', 'wp_template', {
+				resolved: true,
+				slug: data.post_name,
+			} )
+		 )[ 0 ];
+	}
+
+	return newTemplate;
+}
+
+findTemplate.siteUrl = '';
```

Comment:
What's this for?

Reply:
This file is copied as is from edit-site and I believe it's very bad code :P, this should move to `core-data` (selector + resolver)